gobblin-hive-registration/src/main/java/gobblin/hive/spec/SimpleHiveSpec.java

```diff
@@ -30,26 +39,49 @@
  */
 @Getter
 @Alpha
-public class SimpleHiveSpec implements HiveSpec {
+@Builder(builderClassName = "Builder")
+public class SimpleHiveSpec implements HiveSpec, HiveSpecWithPreActivities, HiveSpecWithPostActivities,
+    HiveSpecWithPredicates {
 
   protected final Path path;
   protected final HiveTable table;
   protected final Optional<HivePartition> partition;
+  @Singular @Getter
+  protected final Collection<Activity> preActivities;
+  @Singular @Getter
+  protected final Collection<Activity> postActivities;
+  @Singular @Getter
+  protected final Collection<Predicate<HiveRegister>> predicates;
 
   protected SimpleHiveSpec(Builder<?> builder) {
     this.path = builder.path;
     this.table = builder.table;
-    this.partition = builder.partition;
+    this.partition = builder.partition != null ? builder.partition : Optional.<HivePartition>absent();
+    this.preActivities = builder.preActivities;
+    this.postActivities = builder.postActivities;
+    this.predicates = builder.predicates;
+  }
+
+  private static Builder<Builder<?>> builder() {
+    return new Builder<>();
+  }
+
+  private static Builder<Builder<?>> builder(Path path) {
+    return new Builder<>(path);
   }
 
-  @Getter
   public static class Builder<T extends Builder<?>> {
-    protected final Path path;
-    protected HiveTable table;
-    protected Optional<HivePartition> partition = Optional.absent();
+    private Path path;
+
+    private Builder() {
```

Comment:
Because of lombok. Otherwise lombok would create a public, no arguments, constructor.